src/pkg/git/github.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) CloudCover
+// SPDX-License-Identifier: Apache-2.0
+
+package git
+
+import (
+	"context"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+type GitClient struct {
+	client *github.Client
+}
+
+func GithubClient(token string) Git {
```

Comment:
if we just want the token here, then we should change the config defaults to take just github.token instead of github.username & github.password.